pkg/tsdb/cloudwatch/mocks/cloudwatch_metric_api.go

```diff
@@ -56,3 +57,14 @@ func chunkSlice(slice []*cloudwatch.Metric, chunkSize int) [][]*cloudwatch.Metri
 
 	return chunks
 }
+
+type MetricsClient struct {
```

Comment:
Can't you use `FakeMetricsAPI` from above instead?

Reply:
I'd prefer to use the testify/mock methods (see the tests in this PR) if we are not opposed.

I think the existing tests should be re-written to use the testify/mock version of the mock client and the old mock types should be deleted, but that this could be done later.

What do you think?

Reply:
I think it's not ideal to have two different mock api implementations at the same time. But if the plan is to change that in a PR in the near future it's fine.

Reply:
Thanks.

I think the way that this will go is that the grouping-by-region test will soon be replaced by a test on a smaller function with less responsibility. Then these mock clients I've introduced won't be used and can be deleted.

However, moving forward and trying to add new tests, I do think we should try to use a mock library, especially when our current mock implementation limits us. If we don't want to be hampered by our current mocks and we don't want to have two different implementations, then I vote for prioritizing the use of a mock library and adapting the existing tests and mocks.

Reply:
That's okay